### PR TITLE
Sonar configuration missing in Gradle

### PR DESCRIPTION
**Pull request by jonashendrickx**
### Description

We were missing some configuration for Sonarqube in Gradle.

### Shape

n/a

### Screenshots

n/a

### Checklist
I did the following to ensure that my changes were tested thoroughly:
- __

I did the following to ensure that my changes do not introduce security vulnerabilities:
- __

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -60,3 +61,19 @@ dependencies {
...
     kapt(libs.hilt.compiler)
 }
+
+sonar {
+    properties {
+        property("sonar.projectKey", "bitwarden_passwordless-android")
+        property("sonar.organization", "bitwarden")
+        property("sonar.host.url", "https://sonarcloud.io")
+        property("sonar.sources", "app/src/main/,passwordless/src/main/")
+        property("sonar.tests", "app/src/test/,passwordless/src/test/")
```
Comment (withinfocus) [inline app/build.gradle.kts:71]: ❌ You should define Sonar inclusions and exclusions too.

File: app/build.gradle.kts [hunk 2]
```diff
@@ -60,3 +61,19 @@ dependencies {
...
+    }
+}
+
+tasks {
+    getByName("sonarqube") {
+        dependsOn("check")
+    }
+}
```
Comment (withinfocus) [inline app/build.gradle.kts:79]: ⛏️ End in a newline.

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -47,4 +48,5 @@ hilt-android = { group = "com.google.dagger", name = "hilt-android", version.ref
 hilt-compiler = { group = "com.google.dagger", name = "hilt-android-compiler", version.ref = "hilt" }
 
 [plugins]
-detekt = { id = "io.gitlab.arturbosch.detekt", version.ref = "detekt" }
\ No newline at end of file
+detekt = { id = "io.gitlab.arturbosch.detekt", version.ref = "detekt" }
+sonarqube = { id = "org.sonarqube", version.ref = "sonarqube" }
```
Comment (withinfocus) [inline gradle/libs.versions.toml:52]: ⛏️ Newline.

File: passwordless/build.gradle.kts [hunk 4]
```diff
@@ -160,3 +161,19 @@ signing {
...
     sign(publishing.publications)
 }
+
+sonar {
+    properties {
+        property("sonar.projectKey", "bitwarden_passwordless-android")
+        property("sonar.organization", "bitwarden")
+        property("sonar.host.url", "https://sonarcloud.io")
+        property("sonar.sources", "passwordless/src/main/")
+        property("sonar.tests", "passwordless/src/test/")
```
Comment (withinfocus) [inline passwordless/build.gradle.kts:171]: ❌ Same inclusions and exclusions.

**Review by withinfocus — CHANGES_REQUESTED** [review]

File: app/build.gradle.kts [hunk 5]
```diff
@@ -60,3 +61,21 @@ dependencies {
...
     kapt(libs.hilt.compiler)
 }
+
+sonar {
+    properties {
+        property("sonar.projectKey", "bitwarden_passwordless-android")
+        property("sonar.organization", "bitwarden")
+        property("sonar.host.url", "https://sonarcloud.io")
+        property("sonar.sources", "app/src/main/,passwordless/src/main/")
+        property("sonar.tests", "app/src/test/,passwordless/src/test/")
```
Comment (withinfocus) [inline app/build.gradle.kts:71]: 🎨 The "passwordless" module wouldn't need to be here since it's listed separately below.

**Review by withinfocus — CHANGES_REQUESTED** [review]
Just one item I think!

**Review by withinfocus — APPROVED** [review]

**Review by jrmccannon — APPROVED** [review]

**Pull request merged**